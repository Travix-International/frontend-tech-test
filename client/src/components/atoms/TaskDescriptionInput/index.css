.task-description-input {
  background-color: transparent;
  border: 0;
  color: #545454;
  flex: 1;
  font-size: 20px;
  font-weight: 500;
  line-height: 1.3;
  overflow: hidden;
  padding: 0;
  text-overflow: ellipsis;
  white-space: nowrap; }
