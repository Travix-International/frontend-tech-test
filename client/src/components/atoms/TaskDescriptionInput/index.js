import React from 'react'
import PropTypes from 'prop-types'
import './index.css'

import withSubmitHandler from '../../hocs/withSubmitHandler'

const TaskDescriptionInput = ({ value, handleChange, handleSubmit }) => {
  return (
    <input
      className='task-description-input'
      placeholder='Type the task description...'
      value={value}
			onChange={handleChange}
      onKeyPress={handleSubmit}
    />
  )
}

TaskDescriptionInput.propTypes = {
  value: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
}

export default withSubmitHandler(TaskDescriptionInput)
