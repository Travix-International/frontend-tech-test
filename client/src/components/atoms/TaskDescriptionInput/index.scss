@import 'src/styles/base/colors.scss';
@import 'src/styles/base/fonts.scss';
@import 'src/styles/base/input.scss';

.task-description-input {
  @extend .input;
  
  color: lighten($black, 33%);
  font-size: $font-small;
  font-weight: $font-normal;
}
