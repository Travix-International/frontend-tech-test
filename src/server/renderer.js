import React from 'react';
import ReactServer from 'react-dom/server';
import { StaticRouter as Router, matchPath } from 'react-router';
import { Provider } from 'react-redux';
import { Helmet } from 'react-helmet';
import { /* get as _get, */ isFunction as _isFunction } from 'lodash';
import renderFullPage from './layout';
import configureStore from '../configureStore';
import { entryRouteComponentMap, getEntryAndRoute, entryReducerMap } from './entryAndRoute';

function applyInitStore(req) {
  const entryRouteInfo = getEntryAndRoute(req.path);
  const currentEntry = entryRouteInfo.entry;
  const rootReducer = entryReducerMap[currentEntry];
  const store = configureStore({}, rootReducer);
  req.reduxStore = store;
  req.entryRouteInfo = entryRouteInfo;
  // next();
}

function applyRouteCheckResult(req) {
  const { path, reduxStore: store, query, entryRouteInfo } = req;
  const promises = [];
  const currentEntry = entryRouteInfo.entry;
  const route = entryRouteInfo.route;
  const match = matchPath(path, route);
  const { loadData } = route;

  // set currentEntry to req
  req.currentEntry = currentEntry;

  // it's possible no loadData set
  if (_isFunction(loadData)) {
    let action = loadData(match, query);
    if (action) {
      if (_isFunction(action) || action.type) {
        action = store.dispatch(action);
        if (action.then) {
          promises.push(action);
        }
      }
    }
  }

  return Promise.all(promises);
}

function responsePage(req, res, clientStats) {
  const { url, reduxStore: store, currentEntry } = req;
  const routerContext = {};
  const reduxStateString = JSON.stringify(store.getState()).replace(/</g, '\\u003c');

  const RouteComponent = entryRouteComponentMap[currentEntry];

  const content = (
    <Provider store={store}>
      <Router context={routerContext} location={url}>
        <RouteComponent />
      </Router>
    </Provider>
  );

  // render to sting to get helmet setting
  const contentString = ReactServer.renderToString(content);
  const head = Helmet.renderStatic();
  const htmlString = renderFullPage(contentString, reduxStateString, head, currentEntry, clientStats);

  res.send(htmlString);
}

export default function renderer({ clientStats }) {
  return (req, res) => {
    applyInitStore(req);
    return applyRouteCheckResult(req).then(() => {
      return responsePage(req, res, clientStats);
    });
  };
}
