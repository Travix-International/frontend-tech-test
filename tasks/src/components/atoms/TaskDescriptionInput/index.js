import React from 'react'
import PropTypes from 'prop-types'

const TaskDescriptionInput = ({ value, handleChange }) => {
  return (
    <input
      value={value}
			onChange={handleChange}
    />
  )
}

TaskDescriptionInput.propTypes = {
  value: PropTypes.string,
  handleChange: PropTypes.func.isRequired
}

export default TaskDescriptionInput
