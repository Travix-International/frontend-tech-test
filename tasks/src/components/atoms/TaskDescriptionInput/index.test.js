import React from 'react'
import renderer from 'react-test-renderer'
import TaskDescriptionInput from './index'

describe('TaskDescriptionInput', () => {
  it('should render correctly', () => {
    const handleChange = jest.fn()
    const handleSubmit = jest.fn()
    const tree = renderer.create(
      <TaskDescriptionInput
        value='A description'
        handleChange={handleChange}
        handleSubmit={handleSubmit}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
