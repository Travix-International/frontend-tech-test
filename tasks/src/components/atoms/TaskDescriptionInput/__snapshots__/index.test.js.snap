// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDescriptionInput should render correctly 1`] = `
<input
  onChange={[Function]}
  value="A description"
/>
`;
